### ISSUE TITLE
aadHttpClientFactory uses old refresh tokens in Teams Desktop Client

### ISSUE THREAD

**Issue by wizneeraj** [SharePoint/sp-dev-docs#8611]
### What type of issue is this?

Question

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Target SharePoint environment

SharePoint Online

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [ ] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [X] other (enter in the "Additional environment details" area below)

### Additional environment details

- browser version : **Microsoft Teams Desktop Client**
- SPFx version: v1.10.0
- Node.js version: v10.13.0
- etc

### Issue description

We have a SPFx Application built which uses aadHttpClientFactory to consume Azure AD protected Function Apps. The SPFx application works fine in SharePoint pages. We have created a Teams Tab using Teams App Studio. This Teams Tab just displays the SharePoint Page where the SPFx application is hosted.

We have an issue where the API calls are failing with the below error for some users 

_AADSTS70043: The refresh token has expired or is invalid due to sign-in frequency checks by conditional access. The token was issued on 2022-11-21T14:40:56.1540000Z and the maximum allowed lifetime for this request is 10800.
Trace ID: b0b9a151-8384-6406-8f49-bae28bddff00
Correlation ID: 3d017ca0-7ab3-d000-9c57-7401f6hj615e
Timestamp: 2022-11-25 08:28:28Z_

Teams Desktop client uses old cached tokens and not getting refreshed even though the SharePoint page containing the SPFx component loads fine.

**Comment by AJIXuMuK** [comment]
@wizneeraj - thanks for reporting it!
Few questions.
1. Why did you create a custom app to display a SharePoint page instead of using OOB SharePoint app?
2. If you use OOB SharePoint app - will your web part work correctly?

**Comment by AJIXuMuK** [comment]
@wizneeraj - could you please clarify at what point are you seeing this error?
Also - is there anything specific about the users that are getting the issue?

**Comment by wizneeraj** [comment]
@AJIXuMuK - We see this issue for users when Conditional Access policy forces the users to reauthenticate. Once that happens, the issue will occur for the user, since it seems like either MS teams Desktop client is unable to provide the new Auth token or aadHttpClientFactory is not refreshing the token

**Comment by wizneeraj** [comment]
@lucabandMSFT - I have shared the teams manifest with Maria and she will be able to share it with you.

**Comment by lucabandMSFT** [comment]
hey @wizneeraj ,
I took a look at the manifest; it looks like the manifest doesn't follow the rules described here: https://learn.microsoft.com/en-us/sharepoint/dev/spfx/deployment-spfx-teams-solutions

The key part is to ensure the {teamSiteDomain} moniker is there instead of the url of the tenant. Please, when you have time, let me know if this resolves the issue.

**Comment by wizneeraj** [comment]
I have already tried using the {teamSiteDomain} moniker and it did not work. To reproduce the issue, I have updated the Teams manifest with this moniker and the app version, to ensure we are testing in the latest version. Even now, i am able to replicate the issue. Also, the issue is not with the Teams Personal app, even if I use the SharePoint OOTB App in Teams and point it to the SharePoint Page containing the SPFx component, I am able to reproduce the issue.
